Fix tests that use `eventSender.mouseScrollBy()` to use UIHelper.statelessMouseWheelScrollAt

smfr opened the pull request:
#### d5823dca0a47f5a8bf31732ca7c74b4dd9b6a991
<pre>
Fix tests that use `eventSender.mouseScrollBy()` to use UIHelper.statelessMouseWheelScrollAt
<a href="https://bugs.webkit.org/show_bug.cgi?id=250355">https://bugs.webkit.org/show_bug.cgi?id=250355</a>
rdar://104058894

Reviewed by Ryosuke Niwa.

Replace calls to `eventSender.mouseMoveTo(); eventSender.mouseScrollBy()` with a
call to `UIHelper.statelessMouseWheelScrollAt()`, making things async where necessary.

Add `UIHelper.waitForEventHandler()`, which wraps the given function in an event
handler and waits for it to fire.

* LayoutTests/fast/events/platform-wheelevent-with-delta-zero-crash.html:
* LayoutTests/fast/events/remove-child-onscroll.html:
* LayoutTests/fast/events/scroll-in-scaled-page-with-overflow-hidden.html:
* LayoutTests/fast/events/wheel/platform-wheelevent-in-scrolling-div.html:
* LayoutTests/fast/events/wheel/wheel-event-outside-body.html:
* LayoutTests/fast/events/wheel/wheelevent-basic.html:
* LayoutTests/fast/events/wheel/wheelevent-in-text-node.html:
* LayoutTests/fast/events/wheel/wheelevent-mousewheel-interaction.html:
* LayoutTests/fast/forms/search/search-scroll-hidden-decoration-container-crash.html:
* LayoutTests/fast/repaint/overflow-auto-in-overflow-auto-scrolled.html:
* LayoutTests/fast/scrolling/mac/horizontal-overflow-trapping-small-deltas.html:
* LayoutTests/fast/scrolling/mac/rubberband-overflow-in-wheel-region-root-jiggle.html:
* LayoutTests/fast/scrolling/mac/vertical-scroll-in-horizontal-scroller.html:
* LayoutTests/fast/scrolling/rtl-scrollbars-listbox-scroll.html:
* LayoutTests/fast/scrolling/scrolling-inside-scrolled-overflowarea.html:
* LayoutTests/resources/ui-helper.js:
(window.UIHelper.async statelessMouseWheelScrollAt):
(window.UIHelper.waitForEventHandler):
(window.UIHelper.callFunctionAndWaitForEvent):
* LayoutTests/scrollbars/scroll-rtl-or-bt-layer.html:

Canonical link: <a href="https://commits.webkit.org/259295@main">https://commits.webkit.org/259295@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/017887a5a561f075d93550c8234680697d5c21e7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104477 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13556 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37383 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113754 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173978 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14661 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4475 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96817 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112721 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110244 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11300 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94357 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38896 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93165 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25967 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80565 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6914 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27324 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7039 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3882 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13071 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46884 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6408 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8833 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on LayoutTests/fast/events/wheel/wheelevent-in-text-node.html:
```diff
@@ -2,33 +2,24 @@
-        function wheelHandler(e) {
-            window.theEvent = e;
+            
+            let div = document.querySelector('div');
```
const?

rniwa commented on LayoutTests/scrollbars/scroll-rtl-or-bt-layer.html:
```diff
@@ -24,79 +24,81 @@
-
-    eventSender.mouseScrollBy(1, 0);
+    await Promise.all([
+        UIHelper.waitForEventHandler(rtl, 'scroll', () => {
```
Why not just:
```js
async () => {
    const event = await UIHelper.waitForEventHandler(~);
    ~
}
```

smfr committed:
Fix tests that use `eventSender.mouseScrollBy()` to use UIHelper.statelessMouseWheelScrollAt
https://bugs.webkit.org/show_bug.cgi?id=250355
rdar://104058894

Reviewed by Ryosuke Niwa.

Replace calls to `eventSender.mouseMoveTo(); eventSender.mouseScrollBy()` with a
call to `UIHelper.statelessMouseWheelScrollAt()`, making things async where necessary.

Add `UIHelper.waitForEventHandler()`, which wraps the given function in an event
handler and waits for it to fire.

* LayoutTests/fast/events/platform-wheelevent-with-delta-zero-crash.html:
* LayoutTests/fast/events/remove-child-onscroll.html:
* LayoutTests/fast/events/scroll-in-scaled-page-with-overflow-hidden.html:
* LayoutTests/fast/events/wheel/platform-wheelevent-in-scrolling-div.html:
* LayoutTests/fast/events/wheel/wheel-event-outside-body.html:
* LayoutTests/fast/events/wheel/wheelevent-basic.html:
* LayoutTests/fast/events/wheel/wheelevent-in-text-node.html:
* LayoutTests/fast/events/wheel/wheelevent-mousewheel-interaction.html:
* LayoutTests/fast/forms/search/search-scroll-hidden-decoration-container-crash.html:
* LayoutTests/fast/repaint/overflow-auto-in-overflow-auto-scrolled.html:
* LayoutTests/fast/scrolling/mac/horizontal-overflow-trapping-small-deltas.html:
* LayoutTests/fast/scrolling/mac/rubberband-overflow-in-wheel-region-root-jiggle.html:
* LayoutTests/fast/scrolling/mac/vertical-scroll-in-horizontal-scroller.html:
* LayoutTests/fast/scrolling/rtl-scrollbars-listbox-scroll.html:
* LayoutTests/fast/scrolling/scrolling-inside-scrolled-overflowarea.html:
* LayoutTests/resources/ui-helper.js:
(window.UIHelper.async statelessMouseWheelScrollAt):
(window.UIHelper.waitForEventHandler):
(window.UIHelper.callFunctionAndWaitForEvent):
* LayoutTests/scrollbars/scroll-rtl-or-bt-layer.html:

Canonical link: https://commits.webkit.org/259295@main
